app/controllers/profiles_controller.ts

```diff
@@ -0,0 +1,167 @@
+        message: 'Error creating profile',
+        error: error.message,
+      })
+    }
+  }
+
+  // View all users
+  public async view({ auth, response }: HttpContext) {
+    try {
+      const token = auth.user?.currentAccessToken.identifier
+      if (!token) {
+        return response.badRequest({ message: 'Token not found' })
+      }
+      const profiles = await Profile.query()
+
+      if (profiles.length === 0) {
+        return response.status(404).json({
+          message: 'No profiles found',
+        })
+      }
+
+      return response.status(200).json({
+        data: profiles.map((profile) => ({
+          name: profile.name,
+          email: profile.email,
+          gender: profile.gender,
+          dateOfBirth: profile.dateOfBirth?.toISODate(), // Handle nullable fields safely
+        })),
+      })
+    } catch (error) {
+      return response.status(500).json({
+        message: 'Error retrieving profiles',
+        error: error.message,
+      })
+    }
+  }
+
+  // Update user profile
+  public async update({ auth, request, response }: HttpContext) {
+    try {
+      const token = auth.user?.currentAccessToken.identifier
+      if (!token) {
+        return response.badRequest({ message: 'Token not found' })
+      }
+      const { name, mobile, email, gender, dateOfBirth } = request.only([
+        'name',
+        'mobile',
+        'email',
+        'gender',
+        'dateOfBirth',
+      ])
+
+      const existingUser = await User.findBy('email', email)
+      if (!existingUser) {
+        return response.status(409).json({
+          message: 'Provided email does not exist',
+        })
+      }
+
+      const profile = await Profile.query().where('userId', existingUser.id).first()
+
+      if (!profile) {
+        return response.status(404).json({
+          message: 'Profile not found',
+        })
+      }
+
+      profile.name = name
+      profile.mobile = mobile
+      profile.email = email
+      profile.gender = gender
+      profile.dateOfBirth = DateTime.fromISO(dateOfBirth)
+
+      await profile.save()
+
+      return response.status(200).json({
+        message: 'Profile updated successfully',
+        profile,
+      })
+    } catch (error) {
+      return response.status(500).json({
+        message: 'Error updating profile',
+        error: error.message,
+      })
+    }
+  }
+
+  // Delete user profile
+  public async delete({ auth, request, response }: HttpContext) {
+    try {
+      const token = auth.user?.currentAccessToken.identifier
+      if (!token) {
+        return response.badRequest({ message: 'Token not found' })
+      }
+      const { mobile } = request.only(['mobile'])
+      const existingUser = await Profile.findBy('mobile', mobile)
+      if (!existingUser) {
+        return response.status(409).json({
+          message: 'Provided mobile does not exist',
+        })
+      }
+
+      const profile = await Profile.query().where('userId', existingUser.id).first()
+
```

Comment:
```suggestion
      const existingUser = await Profile.findBy('mobile', mobile)
      if (!existingUser) {
        return response.status(409).json({
          message: 'Provided mobile does not exist',
        })
      }

      const profile = await Profile.query().where('userId', existingUser.id).first()

```

existing User is actual the profile here. How will you find a profile if you try to pass id against userId